daemon/containerd/image_delete.go

```diff
@@ -35,6 +41,24 @@ import "github.com/docker/docker/api/types"
 // If prune is true, ancestor images will each attempt to be deleted quietly,
 // meaning any delete conflicts will cause the image to not be deleted and the
 // conflict will not be reported.
-func (i *ImageService) ImageDelete(imageRef string, force, prune bool) ([]types.ImageDeleteResponseItem, error) {
-	panic("not implemented")
+func (i *ImageService) ImageDelete(ctx context.Context, imageRef string, force, prune bool) ([]types.ImageDeleteResponseItem, error) {
+	// TODO(thaJeztah): implement ImageDelete "force" options; see https://github.com/moby/moby/issues/43850
+	// TODO(thaJeztah): implement ImageDelete "prune" options; see https://github.com/moby/moby/issues/43849
+
+	records := []types.ImageDeleteResponseItem{}
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

Reply:
Hm, yeah, I don't _always_ agree with that rule (sometimes it's really more convenient to initialise it if later on we need to `append()`, and `make(...)` is a bit cumbersome (or sometimes overdoing it).

That said; it looks like (at least currently) it'll only ever create a single instance, so I don't think we need this variable at all; we can just return `[]types.ImageDeleteResponseItem{}{d}` at the end of the function.

Reply:
> sometimes it's really more convenient to initialise it if later on we need to `append()`

Hmm? `append()` works on `nil` slices, same as non-`nil` but empty slices. You might be thinking of maps, where assigning to a `nil` map panics.

Reply:
Yes, I just edited, indeed had maps in mind 🙈